aiosmtpd/smtp.py

```diff
@@ -591,6 +593,32 @@ def _syntax_available(self, method):
         status = await self._call_handler_hook('RSET')
         await self.push('250 OK' if status is MISSING else status)
 
+    def _parse_xclient_attributes(self, attributes):
+        """
+        Parse key-value arguments and return OrderedDict on success.
+        Raise ValueError on wrong arguments.
+        """
+        result = collections.OrderedDict()
+        for kv in attributes.split(' '):
+            key, value = kv.split('=')
+            if key not in XCLIENT_ATTRIBUTES:
+                raise ValueError()
+            result[key] = value
+        return result
+
+    @syntax('XCLIENT [key1=value1] ... [keyN=valueN]')
+    async def smtp_XCLIENT(self, arg):
+        if not hasattr(self.event_handler, 'handle_XCLIENT'):
+            await self.push('550 feature not enabled')
+            return
+        try:
+            attributes = self._parse_xclient_attributes(arg)
+        except ValueError:
```

Comment:
If you do add additional detail to the exception, you can plumb that through to the client.